### PR TITLE
feat(trackLinkClicked): Add a prop to track any link on the jobCard

### PR DESCRIPTION
**Pull request by kosanna**
We need to track clicks on the JobCard links in Omniture.

Previously taken approach was adding `onShelfSectionLinkClick` which implies that for every next link type we will need to add a separate function, eg. `onSomeLinkClick`, which is not an easily extendable API.

In this PR  generic prop `trackLinkClicked` is added and applied to the ShelfSection links.
Every time when shelf link is clicked `trackLinkClicked` function is called with a required shelf link type provided in the component.

To test it open a console and click shelf links, you will see logs for the call with argument of a link type.

This way we can supply any link type inside of the JobCard to `trackLinkClicked`.

Externally provided `trackLinkClicked` should be a function that accepts the link type that needs to be tracked in omniture as a first argument, eg.

```js
const trackLinkClicked = linkType => {
  // ...do something with linkType
}
```

- Plus some general clenup on props

### CODE REVIEW

**Comment by kosanna** [comment]
Shutting down as probably will move it to srp-react

File: react/JobCard/components/ShelfSection/ShelfSection.js [hunk 1]
```diff
@@ -1,81 +1,81 @@
...
+          const trackOmniture = partial(trackLinkClicked, shelfItem.searchMethod);
+          const rednerLink = linkObject => (
+            <LinkComponent
+              key={linkObject.name}
+              {...linkObject}
+              className={styles.shelfLink}
+              onClick={trackOmniture}>
+              {linkObject.name}
+            </LinkComponent>
+          );
```
Comment (hscgavin) [inline react/JobCard/components/ShelfSection/ShelfSection.js]: Can we move `trackOmniture` and `rednerLink` functions out of `map`? An instance of a function is created for each iteration of the loop

Comment (kosanna) [reply react/JobCard/components/ShelfSection/ShelfSection.js]: Yeah, I know right, was trying to stop going down the rabbit hole 🤕 We will still need to define the render here to wire it up to enchanted omniture tracking type which is available only within the map 🤕 But I can try to make it slightly better :)

File: react/JobCard/components/ShelfSection/ShelfSection.js [hunk 2]
```diff
@@ -1,81 +1,82 @@
...
+  );
+
+  return (
+    <Section className={classnames(styles.root, { [styles.showShelfSection]: showShelfSection })}>
+      <div className={styles.shelfDivider} />
+      {shelfLinks && shelfLinks.length &&
       <Text whispering className={styles.shelfLinksContainer}>
-        {shelfLinks.map((shelfItem, i) => {
+        {shelfLinks.map(shelfItem => {
+          const trackOmniture = partial(trackLinkClicked, shelfItem.searchMethod);
```
Comment (cm-seekasia) [inline react/JobCard/components/ShelfSection/ShelfSection.js]: It's good. Just another suggestion without 3rd party lib(lodash):
```
const trackOmniture = () => trackLinkClicked(shelfItem.searchMethod);
```

**Pull request merged**